src/main/java/com/renatusnetwork/momentum/data/blackmarket/BlackMarketManager.java

```diff
@@ -299,13 +315,15 @@ public void playerLeft(PlayerStats playerStats, boolean disconnected)
             player.sendMessage(Utils.translate(Momentum.getSettingsManager().blackmarket_message_prefix + " &cYou have missed the opportunity of a lifetime."));
     }
 
-    public boolean isRunning()
-    {
-        return running != null;
-    }
-
+    public boolean isRunning() { return running != null; }
+    public boolean isInPreparation() { return inPreparation; }
     public BlackMarketEvent getRunningEvent() { return running; }
 
+    // check if running before calling
+    public long getTimeBeforeStart() {
+        return Math.max(300000 - (System.currentTimeMillis() - startTime), 0); // 300000 ms = 5 minutes
```

Comment:
Little micro-optimization would be to do?: return (startTime + 300000) - System.currentTimeMillis();

Reply:
itd be the same thing, no? or is it more optimized to do it that way for some reason?

Reply:
We don't need the math.max in that way. It's very minor but if this is gonna be run every 2 ticks for each player it would be impactful by a few ms

Reply:
oh i see what you mean. i forgot the reason i put the math.max there to begin with.